fe/fe-core/src/main/java/org/apache/doris/analysis/InsertStmt.java

```diff
@@ -304,7 +304,9 @@ public void analyze(Analyzer analyzer) throws UserException {
 
         db = analyzer.getEnv().getCatalogMgr().getCatalog(tblName.getCtl()).getDbOrAnalysisException(tblName.getDb());
         // create label and begin transaction
-        long timeoutSecond = ConnectContext.get().getExecTimeout();
+        ConnectContext ctx = ConnectContext.get();
+        int timeoutSecond = Math.max(ctx.getExecTimeout(), ctx.getSessionVariable().getInsertTimeoutS());
+        ctx.setExecTimeout(timeoutSecond);
```

Comment:
Once we call setExecTimeout here, the ConnectContext has already got the required exec_timeout.